File: backend/schemas/collection.py
"""Collection schemas for request/response validation."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CollectionCreate(BaseModel):
    name: str
    description: Optional[str] = None
    scope: str = "personal"  # personal | organization | stream
    stream_id: Optional[int] = None


class CollectionUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class CollectionResponse(BaseModel):
    collection_id: int
    name: str
    description: Optional[str]
    scope: str
    stream_id: Optional[int]
    article_count: int
    created_by: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CollectionArticleAdd(BaseModel):
    article_id: int


class CollectionDetailResponse(CollectionResponse):
    """Collection with its articles included."""
    pass

File: frontend/src/types/collection.ts
export interface Collection {
    collection_id: number;
    name: string;
    description?: string;
    scope: 'personal' | 'organization' | 'stream';
    stream_id?: number;
    article_count: number;
    created_by: number;
    created_at: string;
    updated_at: string;
}

export interface CollectionArticle {
    article_id: number;
    title: string;
    authors: string[];
    journal?: string;
    pmid?: string;
    doi?: string;
    abstract?: string;
    url?: string;
    pub_year?: number;
    pub_month?: number;
    pub_day?: number;
    added_at: string;
    added_by: number;
}
